src/services/state-manager/state-manager.service.ts

```diff
@@ -1,39 +1,54 @@
-import _ from "lodash"
-import { Document } from "mongoose"
-import { IState } from "../../types/global/state"
-import states from "../../data/models/state-schema"
-
-export class StateManagerService {
-  private static _instance: StateManagerService
-
-  public static getInstance() {
-    if (_.isNil(StateManagerService._instance)) {
-      StateManagerService._instance = new StateManagerService()
-    }
-    return StateManagerService._instance
-  }
-
-  public async getBotState(memberId: string): Promise<Document | void> {
-    return await states.findOne({ memberId }, (err: Error, stateFound: Document): IState => {
-      if (stateFound) return stateFound.schema.obj
-    })
-    .catch((err): void => {
-      console.log(`Error: State not found because: ${err}`)
-      return
-    })
-  }
-
-  public setBotState(memberId: string, newState: IState): void {
-    this.getBotState(memberId)
-    .then((stateFound): void => {
-      if (stateFound) {
-        states.update({ memberId }, newState, (err: Error) => {
-          if (err) throw err
-        })
-      }
-      else {
-        states.create(newState)
-      }
-    })
-  }
-}
+import _ from "lodash";
+import mongoose from "mongoose";
+import { IState } from "../../types/global/state";
+import { StateNamesEnum } from "../../enums/state-names.enum";
+
+export class StateManagerService {
+  private static _instance: StateManagerService;
+
+  public static getInstance(): StateManagerService {
+    if (_.isNil(StateManagerService._instance)) {
+      StateManagerService._instance = new StateManagerService();
+    }
+    return StateManagerService._instance;
+  }
+
+  private _currentState: IState = {
+    memberId: `None`,
+    state: {
+      data: ``,
+      name: StateNamesEnum.NORMAL,
+      step: 0,
+    },
+  };
+
+  public async getBotState(memberId: string): Promise<IState | void> {
+    return mongoose
+      .model(`stateSchema`)
+      .findOne({ memberId })
+      .then((foundState: mongoose.Document | null): void => {
+        if (this._isState(foundState)) {
+          console.log(`Character found with name: ${foundState.state.name}`);
+          void Promise.resolve(foundState);

```

Comment:
Nope the void is a really bad place here.
You MUST return the Promise to make your code works.

Reply:
Ok.